lib/mongo/client_encryption.rb

```diff
@@ -79,7 +79,7 @@ def create_data_key(kms_provider, options={})
     # @return [ BSON::Binary ] A BSON Binary object of subtype 6 (ciphertext)
     #   representing the encrypted value
     def encrypt(value, opts={})
-      doc = { 'v': value }
+      doc = BSON::Document.new('v': value)
```

Comment:
Actually this was fine the way it was. The principal issue with Hash vs Document is when using symbol keys and then using the `#keys` method or iterating the hash, since Document stringifies the keys.

The Document way is fine as well, though I imagine Hash is more performant since keys aren't converted.